src/daft-recordbatch/src/file_info.rs

```diff
@@ -81,6 +107,23 @@ impl FileInfos {
     }
 }
 
+impl IntoIterator for FileInfos {
+    type Item = (String, Option<i64>, Option<i64>);
```

Comment:
i would expect the `Item` to be a `FileInfo` here. Since we're only implementing this for the extend function, i'd instead just make this into a helper function.